### PR TITLE
feat(teams): team models & migrations

### PR DESCRIPTION
**Pull request by cfanoulis**
Adds initial team models as well as associating them with users.
Draws inspiration heavily from https://github.com/ChristophWurst/diesel_many_to_many

### CODE REVIEW

File: src/models/team.rs
Comment (cjdenio) [inline src/models/team.rs]: This shouldn't be optional

File: migrations/2021-08-28-153817_create_team_users/down.sql
Comment (cjdenio) [inline migrations/2021-08-28-153817_create_team_users/down.sql]: Should be `DROP TABLE team_users`

File: migrations/2021-08-28-153817_create_team_users/up.sql
Comment (cjdenio) [inline migrations/2021-08-28-153817_create_team_users/up.sql]: Use a composite primary key instead, like so:

```sql
CREATE TABLE team_users (
  user_id INTEGER NOT NULL REFERENCES users (id),
  team_id INTEGER NOT NULL REFERENCES teams (id),
  PRIMARY KEY (user_id, team_id)
)
```

**Review by cjdenio — CHANGES_REQUESTED** [review]
Looks great! Just a few small suggestions

**Comment by cfanoulis** [comment]
@hack-as-a-service/backend take a peek as well pls

also I'll be squashing before you merge so please wait for an all clear prior to merge

File: src/schema.rs
Comment (cjdenio) [inline src/schema.rs]: Reset your database and run `diesel migration run` to update this file

File: src/models/team_user.rs
Comment (cjdenio) [inline src/models/team_user.rs]: Remove this and add `#[primary_key(team_id, user_id)]` above the struct

**Review by cjdenio — CHANGES_REQUESTED** [review]
Just a few quick cleanup things

File: migrations/2021-08-28-150018_create_teams/up.sql
Comment (anirudhb) [inline migrations/2021-08-28-150018_create_teams/up.sql]: This should be `DEFAULT FALSE` since a user will really only have one personal team, so it's better to special-case that as compared to normal team creation

**Review by anirudhb — COMMENTED** [review]
lgtm

**Review by cjdenio — APPROVED** [review]

**Comment by cfanoulis** [comment]
all clear for merge!

**Pull request merged**
